### DISCUSSION TITLE
hoveredRow is null

### DISCUSSION THREAD

**Discussion by kricca** [KevinVandy/material-react-table#941] [General]
I am using the drag-n-drop functionality between two tables. I've followed the example from the website. The majority of time this works fine. After a while, the reordering of rows in a table breaks because the _hoveredRow_ is always null. I'm following the examples and using _getState()_ similar to this:
```
muiRowDragHandleProps: ({ table }) => ({
      onDragEnd: () => {
        const { draggingRow, hoveredRow } = table.getState();
        if (hoveredRow && draggingRow) {
          data.splice(
            (hoveredRow as MRT_Row<Person>).index,
            0,
            data.splice(draggingRow.index, 1)[0],
          );
          setData([...data]);
        }
      },
    }),
```

The _draggingRow_ is populated, but _hoveredRow_ is not. I clear cache, try incognito, different browsers, same results. The only thing that resolves this issue is a reboot, which seems odd. Any idea what could be causing this?

An additional note...even the examples on the MRT website do not work when my PC seems to get in this state. After the reboot if I go back to the examples on the MRT website, they all of a sudden work. I've added this last additional note just to show that it isn't an issue with my implementation, since even the MRT examples do not work.

**Comment by KevinVandy** [comment]
Sorry, it's really hard to help without a reproducible sandbox. I've never seen this bug before in my own implementations. Chances are it's a bug in your own code trying to read the hoveredRow state from the wrong table, but I can't tell unless you shared all of the code.

**Reply by kricca** [reply]
Thanks for the reply Kevin. I was seeing the issue with your examples for row ordering and row dragging, so it wasn't isolated to my code. I was hoping someone would have experienced the same issue. I'll keep digging around.

**Reply by KevinVandy** [reply]
Can you give instructions for how to reproduce? You're saying that dragging rows back and forth between the tables stops working at some point?

**Reply by kricca** [reply]
I wish I could. It's random and there is no event that occurs to trigger this that I can see. I'll just go try to re-order rows and it does not work. To clarify...it is not dragging between the two tables that stops working. It is the re-ordering. As soon as I click the handle to move a row, it immediately switches from one table to the other. If I grab the handle to re-order, it moves it to the opposite table. This is because whatever event triggers the hoveredRow to be set does not happen. I only end up with a draggableRow object. This seems O/S based because when this does occur, it is happening on my local environment, my deployed environment, different browsers, and your website examples. Honestly it could be device related perhaps. I'm using a Logitech MX Master 2S. Maybe whatever event this sends to trigger the hoveredRow fails, which is why I end up needing to reboot. I wish I could provide some sample code, but this seems like an anomaly. I was just hoping someone else experienced the same thing. BTW...this library is awesome!
